src/App.ts

```diff
@@ -1040,6 +1028,145 @@ export default class App {
   }
+
+  // ---------------------
+  // Private methods for initialization
+  // ---------------------
+
+  private initReceiver(
+    receiver?: Receiver,
+    signingSecret?: HTTPReceiverOptions['signingSecret'],
+    endpoints?: HTTPReceiverOptions['endpoints'],
+    port?: HTTPReceiverOptions['port'],
+    customRoutes?: HTTPReceiverOptions['customRoutes'],
+    processBeforeResponse?: HTTPReceiverOptions['processBeforeResponse'],
+    signatureVerification?: HTTPReceiverOptions['signatureVerification'],
+    clientId?: HTTPReceiverOptions['clientId'],
+    clientSecret?: HTTPReceiverOptions['clientSecret'],
+    stateSecret?: HTTPReceiverOptions['stateSecret'],
+    redirectUri?: HTTPReceiverOptions['redirectUri'],
+    installationStore?: HTTPReceiverOptions['installationStore'],
+    scopes?: HTTPReceiverOptions['scopes'],
+    appToken?: string,
+    logger?: Logger,
+  ): Receiver {
+    if (receiver !== undefined) {
+      // Custom receiver supplied
+      if (this.socketMode === true) {
+        // socketMode = true should result in SocketModeReceiver being used as receiver
+        // TODO: Add case for when socketMode = true and receiver = SocketModeReceiver
+        // as this should not result in an error
+        throw new AppInitializationError('You cannot supply a custom receiver when socketMode is set to true.');
+      }
+      return receiver;
+    }
+    if (this.socketMode === true) {
+      if (appToken === undefined) {
+        throw new AppInitializationError('You must provide an appToken when socketMode is set to true. To generate an appToken see: https://api.slack.com/apis/connections/socket#token');
+      }
+      this.logger.debug('Initializing SocketModeReceiver');
+      return new SocketModeReceiver({
+        appToken,
+        clientId,
+        clientSecret,
+        stateSecret,
+        redirectUri,
+        installationStore,
+        scopes,
+        logger,
+        logLevel: this.logLevel,
+        installerOptions: this.installerOptions,
+        customRoutes,
+      });
+    }
+    if (signatureVerification === true && signingSecret === undefined) {
+      // Using default receiver HTTPReceiver, signature verification enabled, missing signingSecret
+      throw new AppInitializationError(
+        'signingSecret is required to initialize the default receiver. Set signingSecret or use a ' +
+          'custom receiver. You can find your Signing Secret in your Slack App Settings.',
+      );
+    }
+    this.logger.debug('Initializing HTTPReceiver');
+    return new HTTPReceiver({
+      signingSecret: signingSecret || '',
+      endpoints,
+      port,
+      customRoutes,
+      processBeforeResponse,
+      signatureVerification,
+      clientId,
+      clientSecret,
+      stateSecret,
+      redirectUri,
+      installationStore,
+      scopes,
+      logger,
+      logLevel: this.logLevel,
+      installerOptions: this.installerOptions,
+    });
+  }
+
+  private initAuthorizeIfNoTokenIsGiven(
```

Comment:
Hmm, in the same vein as `initReceiver` and for the sake of consistency, I'm wondering if we should make this method return an authorize object and then the caller of this method sets the class instance's `authorize` property? This may be my own personal preference and I don't have a strong opinion about this, but I would prefer to see less side effects inside these secondary private methods (i.e. `this.authorize = initAuthorize(stuff)`), just like how `initReceiver` is used in this PR (i.e. `this.receiver = initReceiver(stuff)`).
What does everyone else think?

Reply:
Thanks for the suggestion. Sounds great. Will update this method!

Reply:
Thanks for tackling this!